doc/source/usersguide/tutorials.rst

```diff
@@ -1227,18 +1227,18 @@ can be passed to this method::
         particle = table.row
 
         # Fill the table with 257 particles
-        for i in xrange(257):
+        for i in range(257):
             # First, assign the values to the Particle record
-            particle['name'] = 'Particle: %6d' % (i)
+            particle['name'] = f'Particle: {i:6d}'
             particle['lati'] = i
             particle['longi'] = 10 - i
 
             ########### Detectable errors start here. Play with them!
-            particle['pressure'] = array(i*arange(2*3)).reshape((2,4))  # Incorrect
-            #particle['pressure'] = array(i*arange(2*3)).reshape((2,3)) # Correct
+            particle['pressure'] = np.array(i * np.arange(2 * 3)).reshape((2, 4))  # Incorrect
+            #particle['pressure'] = np.array(i * np.arange(2 * 3)).reshape((2, 3)) # Correct
             ########### End of errors
 
-            particle['temperature'] = (i**2)     # Broadcasting
+            particle['temperature'] = i ** 2     # Broadcasting
 
```

Comment:
Personally I'm not used to see spaces around the ``**`` operator but it is OK

Reply:
Would we consider introducing some sort of automatic formatter for the code, such as `black`? I believe `black` puts spaces there, but with a formatter (and a `pre-commit` rule that would apply all those changes to any commit) there would be no need to take special care of it.

Reply:
Honestly I'm not a big fan of pre-commit hooks the modify the code.
I have always checked the code with [flake8](https://flake8.pycqa.org) if we want to switch to black I'm fine with it but I would prefer to to use this tools offline.

Reply:
They are meant to check whether the code has been formatted/linted properly and ask the contributor to do so. `flake8` is one of the options, of course.